pkg/provider/github/parse_payload.go

```diff
@@ -524,7 +532,42 @@ func (v *Provider) handleIssueCommentEvent(ctx context.Context, event *github.Is
 	}
 
 	v.Logger.Infof("issue_comment: pipelinerun %s on %s/%s#%d has been requested", action, runevent.Organization, runevent.Repository, runevent.PullRequestNumber)
-	return v.getPullRequest(ctx, runevent)
+	pr, err := v.getPullRequest(ctx, runevent)
+	if err != nil {
+		return nil, err
+	}
+
+	commentBody := event.GetComment().GetBody()
+	if opscomments.IsOkToTestComment(commentBody) && v.pacInfo.RequireOkToTestSHA {
+		shaFromCommentRaw := opscomments.GetSHAFromOkToTestComment(commentBody)
+		if shaFromCommentRaw == "" {
+			v.Logger.Errorf(errSHANotProvided)
+			if err := v.CreateComment(ctx, runevent, fmt.Sprintf(errSHANotProvidedComment, pr.SHA), ""); err != nil {
+				v.Logger.Errorf("failed to create comment: %v", err)
+			}
+			return info.NewEvent(), errors.New(errSHANotProvided)
+		}
+		shaFromComment := strings.ToLower(shaFromCommentRaw)
+		prSHALower := strings.ToLower(pr.SHA)
+		if len(shaFromCommentRaw) < 40 && !strings.HasPrefix(prSHALower, shaFromComment) {
+			msg := fmt.Sprintf(errSHAPrefixMismatch, shaFromCommentRaw, pr.SHA)
+			v.Logger.Errorf(msg)
+			if err := v.CreateComment(ctx, runevent, msg, ""); err != nil {
+				v.Logger.Errorf("failed to create comment: %v", err)
+			}
+			return info.NewEvent(), fmt.Errorf(errSHAPrefixMismatch, shaFromCommentRaw, pr.SHA)
+		}
+		if len(shaFromCommentRaw) >= 40 && prSHALower != shaFromComment {
```

Comment:
```suggestion
		if prSHALower != shaFromComment {
```

if length is greater than 40 then its not mismatch error